### ISSUE TITLE
Failed to execute process:startx

### ISSUE THREAD

**Issue by ghost** [fish-shell/fish-shell#1281]
FreeBSD 10
fish, version 2.1.0

```
Failed to execute process '/usr/local/bin/startx'. Reason:
exec: Exec format error
The file '/usr/local/bin/startx' is marked as an executable but could not be run by the operating system.
```

**Comment by krader1961** [comment]
Someone was asking about this on IRC/Gitter recently because they were seeing the same error. Albeit with different scripts created at their company that did not contain a [shebang line](https://en.wikipedia.org/wiki/Shebang_(Unix)). They noticed that those scripts ran just fine when launched from sh/bash and therefore didn't understand why fish reported an error.

The answer is that the Bourne shells and those derived from it like Bash and Zsh contain a compatibility shim that is no longer necessary. There is a common misconception that the shebang line is read by the shell. That isn't true: it's read by the kernel. Specifically, whatever part of the kernel implements the `execve()` call. Long, long, ago (like 35+ years) shebang lines were new and there was no guarantee your kernel would read them to figure out which shell to run to interpret the script. So if the `execve("/path/to/script")` call failed your shell (/bin/sh and /bin/csh back then was pretty much the only options) would just assume the script was a Bourne shell script and directly call `execve("/bin/sh", "/path/to/script")`.

There is a kernel of truth to the notion that the shebang line is read by the shell. Long ago the Bourne shell would look to see if the file began with `:` or `#` characters which was interpreted as meaning the file contained a Bourne shell script. That behavior was made obsolete and ultimately removed after the introduction of shebang lines.

Bottom line is that today you should not rely on that compatibility shim. Every script must have a shebang line. Full stop. Anyone telling you it's okay to omit the shebang line is propagating a dangerous practice.
